src/test/test_hs_service.c

```diff
@@ -2285,14 +2285,9 @@ test_export_client_circuit_id(void *arg)
   cp1 = buf_get_contents(conn->outbuf, &sz);
   tt_str_op(cp1, OP_EQ,
             "PROXY TCP6 fc00::ffff:ffff:ffff:ffff:ffff:ffff ::1 65535 42\r\n");
-  tor_free(cp1);
-  extend_info_free(chosen_exit);
-  or_circ->build_state->chosen_exit = NULL;
 
  done:
   UNMOCK(connection_write_to_buf_impl_);
-  extend_info_free(chosen_exit);
-  or_circ->build_state->chosen_exit = NULL;
```

Comment:
It would also help to add a comment explaining that circuit_free() also frees chosen_exit.

Reply:
Fixed in `06a16fc`.